core/src/test/java/io/grpc/transport/MessageFramerTest.java

```diff
@@ -149,60 +145,134 @@ public void flushIsIdempotent() {
     writePayload(framer, new byte[] {3, 14});
     framer.flush();
     framer.flush();
-    verify(sink).deliverFrame(Bytes.asList(new byte[] {0, 0, 0, 0, 2, 3, 14}), false);
+    verify(sink).deliverFrame(toWriteBuffer(new byte[]{0, 0, 0, 0, 2, 3, 14}), false);
     verifyNoMoreInteractions(sink);
   }
 
   @Test
   public void largerFrameSize() throws Exception {
     final int transportFrameSize = 10000;
-    MessageFramer framer = new MessageFramer(copyingSink, transportFrameSize);
+    MessageFramer framer = new MessageFramer(copyingSink, allocator, transportFrameSize);
     writePayload(framer, new byte[1000]);
     framer.flush();
     verify(sink).deliverFrame(frameCaptor.capture(), eq(false));
-    List<Byte> buffer = frameCaptor.getValue();
+    ByteWritableBuffer buffer = frameCaptor.getValue();
     assertEquals(1005, buffer.size());
-    assertEquals(Bytes.asList(new byte[] {0, 0, 0, 3, (byte) 232}), buffer.subList(0, 5));
+
+    byte data[] = new byte[1005];
+    data[3] = 3;
+    data[4] = (byte) 232;
+
+    assertEquals(toWriteBuffer(data, transportFrameSize), buffer);
     verifyNoMoreInteractions(sink);
   }
 
   @Test
   public void compressed() throws Exception {
     final int transportFrameSize = 100;
-    MessageFramer framer = new MessageFramer(copyingSink, transportFrameSize,
-        MessageFramer.Compression.GZIP);
+    MessageFramer framer =
+            new MessageFramer(copyingSink, allocator, transportFrameSize, Compression.GZIP);
     writePayload(framer, new byte[1000]);
     framer.flush();
     verify(sink).deliverFrame(frameCaptor.capture(), eq(false));
-    List<Byte> buffer = frameCaptor.getValue();
+    ByteWritableBuffer buffer = frameCaptor.getValue();
     // It should have compressed very well.
     assertTrue(buffer.size() < 100);
     // We purposefully don't check the last byte of length, since that depends on how exactly it
     // compressed.
-    assertEquals(Bytes.asList(new byte[] {1, 0, 0, 0}), buffer.subList(0, 4));
-    verifyNoMoreInteractions(sink);
+    assertEquals(1, buffer.data[0]);
+    assertEquals(0, buffer.data[1]);
+    assertEquals(0, buffer.data[1]);
+    assertEquals(0, buffer.data[1]);
+  }
+
+  private static WritableBuffer toWriteBuffer(byte[] data) {
```

Comment:
toWriteBuffer(byte... data) would be more convenient. That does cause some problem with toWriteBuffer(byte[], int), so maybe rename it?